jetty-client/src/main/scala/org/http4s/client/jetty/JettyClient.scala

```diff
@@ -0,0 +1,131 @@
+package org.http4s
+package client
+package jetty
+
+import cats.data.Kleisli
+import cats.effect._
+import cats.effect.implicits._
+import cats.implicits.{catsSyntaxEither => _, _}
+import fs2._
+import fs2.Stream._
+import java.nio.ByteBuffer
+import org.eclipse.jetty.client.HttpClient
+import org.eclipse.jetty.client.api.{Result, Request => JettyRequest, Response => JettyResponse}
+import org.eclipse.jetty.client.util.DeferredContentProvider
+import org.eclipse.jetty.http.HttpFields
+import org.log4s.getLogger
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+import scala.util.control.NonFatal
+
+object JettyClient {
+
+  private val logger = getLogger
+
+  def apply[F[_]](client: HttpClient = new HttpClient())(
+      implicit F: ConcurrentEffect[F],
+      ec: ExecutionContext): F[Client[F]] =
+    F.pure(client)
+      .flatTap(client => F.delay { client.start() })
+      .map(client => {
+        Client(
+          Kleisli { req =>
+            F.async[DisposableResponse[F]] { cb =>
+              toRequest(client, req).send(responseHandler(cb))
+            }
+          },
+          F.delay({
+            try {
+              client.stop()
+            } catch {
+              case NonFatal(t) => logger.warn(t)("Unable to shut down Jetty client")
+            }
+          })
+        )
+      })
+
+  def stream[F[_]](client: HttpClient = new HttpClient())(
+      implicit F: ConcurrentEffect[F],
+      ec: ExecutionContext): Stream[F, Client[F]] =
+    Stream.bracket(apply(client))(_.shutdown)
+
+  private def toRequest[F[_]](client: HttpClient, request: Request[F])(
+      implicit F: Effect[F],
+      ec: ExecutionContext): JettyRequest = {
+    val jReq = client.newRequest(request.uri.toString).method(request.method.name)
+
+    for (h <- request.headers)
+      jReq.header(h.name.toString, h.value)
+
+    jReq.content(requestContentProvider(request))
+  }
+
+  private def requestContentProvider[F[_]](
+      request: Request[F])(implicit F: Effect[F], ec: ExecutionContext) = {
+    val dcp = new DeferredContentProvider()
+
+    ec.execute(new Runnable {
```

Comment:
I'm not sure if running this on the ec used for running the responses is the right thing to do... ideas?

Reply:
so, I don't think that `offer` actually blocks the thread, the chunks are written to a `java.util.ArrayDeque` that automatically resizes when needed (code [here](https://github.com/eclipse/jetty.project/blob/5b5f2fcf5f8241dd9edb998b97ad5e6cd45130a8/jetty-client/src/main/java/org/eclipse/jetty/client/util/DeferredContentProvider.java#L155)) 

the reason why I relocated it is that otherwise the request won't be sent until the entire body has been loaded into the content provider. I could move the execution of this block of code after the `request.send(...)` so that it wouldn't need to be relocated.